src/main/scala/nl.knaw.dans.easy.deposit/package.scala

```diff
@@ -15,34 +15,62 @@
  */
 package nl.knaw.dans.easy
 
-import java.nio.file.{ FileAlreadyExistsException, Paths }
+import java.nio.file.{ Path, Paths }
 import java.util.UUID
 
-import better.files.StringOps
+import better.files.{ File, StringOps }
 import nl.knaw.dans.bag.DansBag
 import nl.knaw.dans.easy.deposit.docs.StateInfo.State.State
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+import org.scalatra.servlet.FileItem
 
 import scala.util.{ Failure, Try }
 import scala.xml._
 
-package object deposit {
+package object deposit extends DebugEnhancedLogging {
 
-  sealed abstract class DepositException(msg: String, cause: Throwable) extends Exception(msg, cause)
-
-  case class NoSuchDepositException(user: String, id: UUID, cause: Throwable)
-    extends DepositException(s"Deposit with id $id not found for user $user:  ${ cause.getMessage }", cause)
+  class ForbiddenException(httpResponseBody: String) extends Exception(httpResponseBody)
+  class BadRequestException(httpResponseBody: String) extends Exception(httpResponseBody)
+  class ConflictException(httpResponseBody: String) extends Exception(httpResponseBody)
+  class NotFoundException(httpResponseBody: String) extends Exception(httpResponseBody)
+  case class OverwriteException(httpResponseBody: String) extends ConflictException(httpResponseBody)
 
   case class CorruptDepositException(user: String, id: String, cause: Throwable)
-    extends DepositException(s"Invalid deposit uuid $id for user $user: ${ cause.getMessage }", cause)
+    extends Exception(s"Invalid deposit uuid $id for user $user: ${ cause.getMessage }", cause)
+
+  case class IllegalStateTransitionException(oldState: State, newState: State)
+    extends ForbiddenException(s"Cannot transition from $oldState to $newState")
+
+  case class IllegalDepositStateException(action: String, actual: State, allowed: Seq[State])
+    extends ForbiddenException(s"Deposit has state $actual, can only $action deposits with one of the states: ${ allowed.mkString(", ") }")
 
-  case class ConcurrentUploadException(tempPrefix: String)
-    extends DepositException(s"Another upload is pending. Please try again later.", new FileAlreadyExistsException(tempPrefix))
+  case class ZipMustBeOnlyFileException(item: FileItem)
+    extends BadRequestException(s"A multipart/form-data message contained a ZIP part [${ item.name }] but also other parts.")
 
-  case class IllegalStateTransitionException(user: String, id: UUID, oldState: State, newState: State)
-    extends DepositException(s"Cannot transition from $oldState to $newState (deposit id: $id, user: $user)", null)
+  /**
+   * Note 1: submit area == ingest-flow-inbox
```

Comment:
That of course depends on the configuration. Might be different in the future (not sure).

```suggestion
   * Note 1: submit area == easy-ingest-flow-inbox
```